pkg/admission/apiconversion/apiconversion_admission.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiconversion
+
+import (
+	"context"
+	"fmt"
+	"io"
+
+	"github.com/kcp-dev/logicalcluster/v3"
+
+	apiextensionsinternal "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions"
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	structuralschema "k8s.io/apiextensions-apiserver/pkg/apiserver/schema"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	"k8s.io/apiserver/pkg/admission"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+
+	"github.com/kcp-dev/kcp/pkg/admission/initializers"
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	kcpinformers "github.com/kcp-dev/kcp/pkg/client/informers/externalversions"
+	"github.com/kcp-dev/kcp/pkg/conversion"
+	"github.com/kcp-dev/kcp/pkg/reconciler/apis/apibinding"
+)
+
+const (
+	PluginName = "apis.kcp.dev/APIConversion"
+)
+
+func Register(plugins *admission.Plugins) {
+	plugins.Register(PluginName,
+		func(_ io.Reader) (admission.Interface, error) {
+			return &apiConversionAdmission{
+				Handler: admission.NewHandler(admission.Create, admission.Update),
+			}, nil
+		})
+}
+
+type apiConversionAdmission struct {
+	*admission.Handler
+
+	getAPIResourceSchema func(clusterName logicalcluster.Name, name string) (*apisv1alpha1.APIResourceSchema, error)
+}
+
+// Ensure that the required admission interfaces are implemented.
+var (
+	_ = admission.ValidationInterface(&apiConversionAdmission{})
+	_ = admission.InitializationValidator(&apiConversionAdmission{})
+	_ = initializers.WantsKcpInformers(&apiConversionAdmission{})
+)
+
+// Validate ensures all the conversion rules specified in an APIConversion are correct.
+func (o *apiConversionAdmission) Validate(ctx context.Context, a admission.Attributes, _ admission.ObjectInterfaces) error {
+	if a.GetResource().GroupResource() != apisv1alpha1.Resource("apiconversions") {
+		return nil
+	}
+
+	cluster, err := genericapirequest.ValidClusterFrom(ctx)
+	if err != nil {
+		return admission.NewForbidden(a, fmt.Errorf("error determining workspace: %w", err))
+	}
+
+	if cluster.Name == apibinding.SystemBoundCRDsClusterName {
+		// TODO(ncdc): do we also want to validate these conversions? They've already been validated once (in their
+		// original logical cluster).
```

Comment:
we won't copy APIConversions into `system:bound-crds`, will we?

Reply:
neither into the bound workspace (I hope 🙏)

Reply:
My current code copies from the source workspace into system:bound-crds, but that is just the initial implementation. Will change it to use what's in the source workspace instead of copying.

Reply:
look into the cache server (and today into the source workspace, yes).

Reply:
Fixed